salsa-family/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "salsa-family"
```

Comment:
If @newpavlov has `salsa20` and `chacha20`, how about using those as the crate names?

Reply:
You could presumably have different round versions the way this crate is written, though.  Also, it would take more work to split out chacha from this crate.

Reply:
I think people are much likelier to find and use these implementations if the crates are called `salsa20` and `chacha20`.

The family name is "The Salsa20 family":

https://cr.yp.to/snuffle/salsafamily-20071225.pdf

> You could presumably have different round versions the way this crate is written, though. 

As it were, the original paper referred to the reduced round variants as "Salsa20/12" and "Salsa20/8".

> Also, it would take more work to split out chacha from this crate.

Do they actually share code anywhere? I was trying to find where if so...

Reply:
There's SalsaFamilyState and SalsaFamilyCipher, which has a bunch of common code for both salsa and chacha.  Salsa20/ChaCha20 are also differentiated from Salsa/ChaCha to make it easier to implement reduced-round versions.

Reply:
How about we call the crate `salsa20` and go from there?

I think it might eventually make sense to have `salsa20`, `chacha20`, and perhaps a third additional crate that can abstract across the differences between them for efficient (i.e. SIMD) implementations. I'd also be happy to submit some proposed ways in which it could be factored apart into different crates.

> Salsa20/ChaCha20 are also differentiated from Salsa/ChaCha to make it easier to implement reduced-round versions.

Yeah, I'm not suggesting you go and name things `Salsa20_12` or `Salsa20_8` or something along those lines. But I think perhaps the impetus for the original "Salsa20/N" naming scheme from the Salsa20 paper was to imply that Salsa20 is really, really what you want to be using, barring extremely unusual circumstances.

Reply:
What do you think about `salsa20-family` as the crate name?

Reply:
I can do that.  I'll make the change over and push it.